homeassistant/components/light/hue.py

```diff
@@ -302,7 +302,7 @@ def __init__(self, light_id, info, bridge, update_lights,
     def unique_id(self):
         """Return the ID of this Hue light."""
         return "{}.{}".format(
-            self.__class__, self.info.get('uniqueid', self.name))
+            self.__class__, self.info.get('uniqueid', self.light_id))
```

Comment:
Interestingly, the API says group names are unique:

> The new name for the group. If the name is already taken a space and number will be appended by the bridge e.g. “Custom Group 1”.

Apparently this doesn't hold true across the different types of groups.

Also, 'uniqueid' is only present on API version 1.3+, which means that using an older bridge could still potentially have a conflict if a light and group have the same name and light_id. Could the suggested fix in that case just be to update the bridge or should we add something to differentiate groups and lights so that we can support the older API too?